versioned/tests/src/main/java/org/projectnessie/versioned/tests/AbstractNamespaceValidation.java

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.tests;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonList;
+import static org.projectnessie.model.CommitMeta.fromMessage;
+import static org.projectnessie.versioned.tests.AbstractVersionStoreTestBase.METADATA_REWRITER;
+import static org.projectnessie.versioned.testworker.OnRefOnly.newOnRef;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.ThrowableAssert.ThrowingCallable;
+import org.assertj.core.api.junit.jupiter.InjectSoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.Namespace;
+import org.projectnessie.versioned.BranchName;
+import org.projectnessie.versioned.Delete;
+import org.projectnessie.versioned.Hash;
+import org.projectnessie.versioned.MergeType;
+import org.projectnessie.versioned.Put;
+import org.projectnessie.versioned.ReferenceConflictException;
+import org.projectnessie.versioned.VersionStore;
+
+@ExtendWith(SoftAssertionsExtension.class)
+public abstract class AbstractNamespaceValidation extends AbstractNestedVersionStore {
```

Comment:
Afaict these tests are included in `AbstractVersionStoreTestBase`, and because of that, are available for both `AbstractDatabaseAdapterVersionStoreTest` and `AbstractVersionStoreTests` (`Persist` tests).

Reply:
Ah, yes, thanks for the clarification. I got confused in test nesting and inheritance :facepalm: